### PR TITLE
[BugFix] BE crashed on invalid JSON input with jsonpath

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] bugfix
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/10804

## Problem Summary(Required) ：

The JSONReader does not return an error when the input is invalid JSON and with jsonpath, leading to the BE crash. 
This PR divides the return error of `JsonFunctions::extract_from_object` as 2 types:
1. NotFound: The key is not found. 
2. DataQualityError: the input is invalid.
When `JsonFunctions::extract_from_object` returns NotFound, the next key will continue to be looked up in the object. When the DataQualityError is returned, the input is valid, and the JSONReader returns the error directly.
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/exec/vectorized/json_scanner.cpp [hunk 1]
```diff
@@ -619,19 +618,25 @@ Status JsonReader::_construct_row(simdjson::ondemand::object* row, Chunk* chunk)
                 row->reset();
                 RETURN_IF_ERROR(add_nullable_column_by_json_object(column, _slot_descs[i]->type(),
                                                                    _slot_descs[i]->col_name(), row, !_strict_mode));
-            } else if (!JsonFunctions::extract_from_object(*row, _scanner->_json_paths[i], &val).ok()) {
-                if (strcmp(column_name, "__op") == 0) {
-                    // special treatment for __op column, fill default value '0' rather than null
-                    if (column->is_binary()) {
-                        column->append_strings(std::vector{Slice{"0"}});
+            } else  {
```
Comment (chaoyli) [inline be/src/exec/vectorized/json_scanner.cpp:621]: Add a test case for this error.

**Comment by rickif** [comment]
run all

File: be/src/exprs/vectorized/json_functions.cpp [hunk 2]
```diff
@@ -138,46 +150,21 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
...
-            auto err = tvalue.get_array().get(arr);
-            if (err) {
-                return Status::InvalidArgument(fmt::format("failed to access field as array, field: {}, err: {}", col,
-                                                           simdjson::error_message(err)));
-            }
+            HANDLE_SIMDJSON_ERROR(tvalue.get_array().get(arr),
+                                  fmt::format("failed to access field as array, field: {}", col));
 
-            size_t sz;
-            err = arr.count_elements().get(sz);
```
Comment (trueeyu) [inline be/src/exprs/vectorized/json_functions.cpp:164]: why remove the `arr.count_elements().get(sz);` check?

File: be/src/exprs/vectorized/json_functions.cpp [hunk 3]
```diff
@@ -118,6 +118,18 @@ Status JsonFunctions::json_path_close(starrocks_udf::FunctionContext* context,
 
 Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const std::vector<SimpleJsonPath>& jsonpath,
                                           simdjson::ondemand::value* value) noexcept {
+#define HANDLE_SIMDJSON_ERROR(err, msg)                                                                            \
+    do {                                                                                                           \
+        const simdjson::error_code& _err = err;                                                                    \
+        const std::string& _msg = msg;                                                                             \
+        if (UNLIKELY(_err)) {                                                                                      \
+            if (_err == simdjson::NO_SUCH_FIELD || _err == simdjson::INDEX_OUT_OF_BOUNDS) {                        \
+                return Status::NotFound(fmt::format("err: {}, msg: {}", simdjson::error_message(_err), _msg));     \
```
Comment (trueeyu) [inline be/src/exprs/vectorized/json_functions.cpp:127]: (msg)

**Review by trueeyu — DISMISSED** [review]

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**